pkg/compactor/shuffle_sharding_grouper.go

```diff
@@ -102,21 +120,43 @@ func (g *ShuffleShardingGrouper) Groups(blocks map[ulid.ULID]*metadata.Meta) (re
 	// which we can parallelly compact.
 	var outGroups []*compact.Group
 
-	i := 0
+	// Check if this compactor is on the subring.
+	// If the compactor is not on the subring when using the userID as a identifier
+	// no plans generated below will be owned by the compactor so we can just return an empty array
+	// as there will be no planned groups
+	onSubring, err := g.checkSubringForCompactor()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to check sub-ring for compactor ownership")
+	}
+	if !onSubring {
+		level.Info(g.logger).Log("msg", "compactor is not on the current sub-ring skipping user", "user", g.userID)
```

Comment:
Do we really want this to be Info rather than Debug? Wouldn't info make the log too noisy? I think the reverse log would be more useful -- a log line print out the `userID` that a specific compactor will work on.